api/server/httputils/errors.go

```diff
@@ -102,3 +109,39 @@ func MakeErrorHandler(err error) http.HandlerFunc {
 		}
 	}
 }
+
+// statusCodeFromGRPCError returns status code according to gRPC error
+func statusCodeFromGRPCError(err error) int {
+	switch grpc.Code(err) {
+	case codes.OK: // code 0
+		return http.StatusOK
+	case codes.InvalidArgument: // code 3
+		return http.StatusBadRequest
+	case codes.DeadlineExceeded: // code 4
+		return http.StatusGatewayTimeout
```

Comment:
This does not seem like the correct code.
This is used when the server is acting as a proxy. In this case something just went really wrong. I think this would just be a 500.